src/lib/chawathe98_matcher.rs

```diff
@@ -316,12 +373,18 @@ impl<T: Clone + Debug + Eq + 'static> MappingStoreGraph<T> {
     /// This overall checks from node exists in the same position in tree 2 (To Node Id)
     /// If it does then true otherwise false.
     pub fn check_tree_1_exists_in_tree2(&self, from_node_ancestor: NodeId<FromNodeId>) -> bool {
+        let mut output = true;
         let node_id_to_node = from_node_ancestor.id();
+        if node_id_to_node >= self.to_arena.borrow().size(){
+            return false;
+        }
         let to_node = NodeId::<ToNodeId>::new(node_id_to_node);
         if self.check_subtree(from_node_ancestor, to_node) {
```

Comment:
This can be simplified to just `self.check_subtree(...)` -- no `return`, no `if`.